source/slides/01_linux_basics.rst

```diff
@@ -0,0 +1,506 @@
+:Tails: Security and privacy focused disto that is livecd/usb based
+:CoreOS: Specialize in massive container deployments
+
+You name it, there's a `distro out there`_!
+
+.. _distro out there: http://lwn.net/Distributions/
+
+Linux Basics
+============
+
+What are users?
+---------------
+
+* You, right now
+
+.. code-block:: bash
+
+    $ whoami    # your username
+    $ who       # who is logged in?
+    $ w         # who is here and what are they doing?
+    $ id        # user ID, group ID, and groups you're in
+
+* Not just people: Apache, Mailman, ntp
+
+Users have
+----------
+
+* Username
+* UID
+* Group
+* Shell
+* Usually (but not always) password
+* Usually (but not always) home directory
+
+Managing users
+--------------
+
+.. code-block:: bash
+
+    $ cat /etc/passwd
+    # username:x:UID:GID:GECOS:homedir:shell
+    $ useradd $USER # vs adduser, the friendly Ubuntu version
+    $ userdel $USER
+    $ passwd
+
+.. figure:: ../_static/xkcd215.png
+    :align: center
+
+.. code-block:: bash
+
+    # GECOS: full name, office number and building, office phone extension,
+    # home phone number (General Electric Comprehensive Operating System)
+    $ chfn # change GECOS information; only works sometimes
+    $ finger # tells you someone's GECOS info
+
+Passwords
+---------
+
+* ``/etc/shadow``, not ``/etc/passwd``
+
+.. code-block:: bash
+
+    test@x230 ~ $ ls -l /etc/ | grep shadow
+    -rw-r-----  1 root shadow   1503 Nov 12 17:37 shadow
+
+    $ sudo su -
+    $ cat /etc/shadow
+    daemon:*:15630:0:99999:7:::
+    bin:*:15630:0:99999:7:::
+    sys:*:15630:0:99999:7:::
+    mail:*:15630:0:99999:7:::
+
+    # name:hash:time last changed: min days between changes: max days 
+    #    between changes:days to wait before expiry or disabling:day of
+    #    account expiry
+
+    $ chage # change when a user's password expires
+
+Root/Superuser
+--------------
+
+* UID 0
+* ``sudo``
+
+.. figure:: ../_static/xkcd149.png
+    :align: center
+
+Acting as another user
+----------------------
+
+.. code-block:: bash
+
+    $ su $USER          # become user, with THEIR password
+    $ su                # become root, with root's password
+    $ sudo su -         # use user password instead of root's
+    $ sudo su $USER     # become $USER with your password
+
+.. figure:: ../_static/xkcd838.png
+    :scale: 80%
+
+If someone has permissions errors:
+    * Check that they or their group owns the files
+    * Check that they have the flag +x to execute
+
+
+What are groups?
+----------------
+
+* Manage permissions for groups of users
+
+.. code-block:: bash
+
+    $ groupadd
+    $ usermod
+    $ groupmod
+    $ cat /etc/group
+        root:x:0:
+        daemon:x:1:
+        bin:x:2:
+        sys:x:3:
+        adm:x:4:
+        tty:x:5:
+    # group name:password or placeholder:GID:member,member,member
+
+What are files?
+---------------
+
+* Nearly everything
+* Files have:
+    * Owner
+    * Permissions
+    * inode
```

Comment:
It might be worth distinguishing between [inode](http://lxr.free-electrons.com/source/include/linux/fs.h#L538) and the [i_ino](http://lxr.free-electrons.com/source/include/linux/fs.h#L559) which is the actual number we often refer to as an inode.